docs/content/architectuur/zgw-in-gegevenslandschap.md

```diff
@@ -0,0 +1,188 @@
+
+* Een zaak is geregistreerd in een ZRC. Aan een zaak gerelateerde objecten (RGBZ) worden ook geregistreerd in de ZRC: betrokkenen, besluiten, klantcontacten, zaakobjecten en eventuele zaaktypespecifieke objecten met zaaktypespecifieke eigenschappen.
+* Vanuit een zaak liggen er verwijzingen naar zaaktype (ZTRC), Document vcq. informatieobject (DRC) en eventuele objecten (in theorie de Basisregistratie, maar vaak Gegevensmagazijn)
+* Voor betrokkene is in theorie ook sprake van een verwijzing (naar BRP en HR, kernregistratie medewerkers, kernregistratie bedrijven, kernregistratie klanten,..), maar in de praktijk wordt dit vaak in de ZRC zelf opgeslagen.
+* De Zaakafhandelcomponent bevat nog veel gegevens:
+    * procesdefinities (hoe moet een proces lopen, welke keuzes zijn er mogelijk);
+    * procesgegevens (waar in het proces zitten we, welke keuzes zijn in het proces gemaakt, wie is de behandelaar, welke gegevens heeft hij gebruikt voor welke processtap, ..)
+    * domein/objectgegevens: gegevens over objecten in het domein waartoe het proces behoort. Bijvoorbeeld voor een Subsidiesysteem de gegevens over een aangevraagde subsidie, doelen die men met subsidies wil stimuleren en de voorwaarden en uitnutting van diverse subsidiepotjes.
+
+* “Aanvraag” mist in dit plaatje. Dat klopt ook. Deze maakt onderdeel uit van de domein/objectgegevens. In de huidige praktijk wordt een aanvraag ook nog vaak (onterecht) met een zaak vereenzelvigd. Waarbij het in de vorm van een Informatieobject als PDFje wordt opgeslagen bij een zaak.
+* Er bestaat in de ZRC geen rechtstreekse relatie tussen het merendeel van de domein/objectgegevens en een zaak (dit is beperkt tot zaakobjecten en zaaktypespecifieke gegevens). Idem voor gedetailleerde procesgegevens en een zaak. M.a.w. in het geval van erg complexe processen, kent de zaak in de ZRC enkel dit proces op hoofdlijnen (de zaakstatussen).
+* Wel liggen deze relaties in de specifieke zaakafhandelcomponent: daarin is (impliciet of expliciet) een proces geconfigureerd (=procesdefinitie (“procestype”), bijvoorbeeld afhandelen evenementensubsidieaanvraag). Het verloop van een specifieke instantie van dat proces volgt dit procestype. En er liggen relaties tussen dit proces en de domein/objectregistratie(s) die bij dit proces horen.
+* De ZAC heeft wel uiteraard weet van welk zaaktype en welke zaak er bij een bepaald procestype en bij een bepaalde procesinstantie horen, maar vanuit een zaak is dit dus niet direct opvraagbaar. Opmerking AK De specifieke ZAC, ja.. Maar de generieke ZAC toch niet?
+
+### 1.3.3. Situatie in een Gegevenslandschap
+
+In een gegevenslandschap komt bovenstaand plaatje er, door het scheiden van proces en data, eenmalige opslag en bevraging ‘bij de bron’ en het verminderen van informatiestromen anders uit te zien (zie figuur 2):
+
+![zgw-in-gegevenslandschap.png](./bestanden/zgw-in-gegevenslandschap.png?raw=true)
+*Figuur 2 - ZGW in een Gegevenslandschap*
```

Comment:
Wie wil er nog zaakafhandelcomponent zijn?
Zou de relatie tussen een ZAC en de gegevenslaag niet wat losser moeten worden? Dus dat de ZAC geen kennis hoeft te hebben van al de API’s die zich eronder bevinden, maar gebruik kan maken van een “moeder” api?

Zou er niet een userstory moeten komen zoiets als “ik als ZAC wil maar gebruik maken van 3  API's om alles rondom een zaak in het gegevenslandschap te regelen, zodat ik geen kennis hoef te hebben van de wijze waarop de datalaag  rondom zaken is georganiseerd en zodat ik ook geen hinder ondervind van een andere indeling van de datalaag

Reply:
@RitaBerghuis Goede vragen!

Zeker omdat er meer (g)ZACs dan ZRCs gebouwd zullen worden door verschillende leveranciers. Een "moeder" API is niet eens zo gek gedacht, je ziet meer composite of gateway APIs die zaken samenvoegen en/of veelgebruikte patronen in een nieuwe API stoppen, bovenop de APIs van componenten. Ik pleit hier overigens niet voor op dit moment maar het is iets wat leeft en dus wel helder moet worden in dit stuk.

Zie bijvoorbeeld: https://developers.redhat.com/blog/2017/08/02/organizing-microservices-modern-integration/
*Het woord Informatiemodel komt er overigens niet in voor en de huidige manier wordt omschreven als "the old ESB days" met de "worst tangled spaghetti code or diagrams", ghehehe.*

Reply:
als je de gegevens uit je afhandelcomponenten wilt halen (en dat wil Common Ground), dan zul je al deze gegevens via een API moeten kunnen benaderen...

Ik zie ook wel composite API's ontstaan. De vraag om een moeder-APi begrijp ik echter niet zo goed...hoe zou die eruit kunnen zien en wat lost het dan op?